tests/ui/debuginfo/dwarf-versions.rs

```diff
@@ -0,0 +1,38 @@
+//@ revisions: zero one two three four five six
+
+//@[zero] compile-flags: -Zdwarf-version=0
+//@[zero] error-pattern: requested DWARF version 0 is not supported
+
+//@[one] compile-flags: -Zdwarf-version=1
+//@[one] error-pattern: requested DWARF version 1 is not supported
+
+//@[two] compile-flags: -Zdwarf-version=2
+//@[two] check-pass
+
+//@[three] compile-flags: -Zdwarf-version=3
+//@[three] check-pass
+
+//@[four] compile-flags: -Zdwarf-version=4
+//@[four] check-pass
+
+//@[five] compile-flags: -Zdwarf-version=5
+//@[five] check-pass
+
+//@[six] compile-flags: -Zdwarf-version=6
+//@[six] error-pattern: requested DWARF version 6 is not supported
+
+//@ compile-flags: -g --target x86_64-unknown-linux-gnu --crate-type cdylib
+//@ needs-llvm-components: x86
+
+// This test verifies the expected behavior of various options passed
+// to `-Zdwarf-version`: 1 & 6 (not supported), 2 - 5 (valid)
```

Comment:
possibly should be
```suggestion
// to `-Zdwarf-version`: 0, 1, and 6 (not supported), 2 - 5 (valid)
```
or
```suggestion
// to `-Zdwarf-version`, as we decode it as a u32 which allows inputs like "0" or "131072",
// even though there is no such DWARF version. We currently only support `2..=5`.
```

Reply:
nit: we generally put test explanation at the top of test, so it's easier to find; it doesn't interfere with compiletest